Framework/Crystal/src/ConnectedComponentLabeling.cpp

```diff
@@ -439,12 +439,11 @@ ClusterTuple ConnectedComponentLabeling::executeAndFetchClusters(
   // Get the keys (label ids) first in order to do the next stage in parallel.
   VecIndexes keys;
   keys.reserve(clusters.size());
-  for (auto &cluster : clusters) {
-    keys.push_back(cluster.first);
-  }
+  std::transform(clusters.cbegin(), clusters.cend(), std::back_inserter(keys),
+                 [](const auto &cluster) { return cluster.first; });
   // Write each cluster out to the output workspace
   PARALLEL_FOR_NO_WSP_CHECK()
-  for (int i = 0; i < static_cast<int>(keys.size()); ++i) { // NOLINT
```

Comment:
I've been doing clang-tidy modernize-loop-convert and compilation seems to fail if you use a range-based loop after a PARALLEL check, which is probably why the // NOLINT was put in. Is // NOLINT tripping up cppcheck or is there a Mantid standards reason for removing it?

Reply:
`cppcheck` ignores such comments. I was deleting stuff too hastily here. I'll put the comment back.